src/classes/UTIL_ACCT_Naming.cls

```diff
@@ -35,296 +35,656 @@
+            //If it finds {!{!FirstName}}
+            setFNSpec = true;
+            firstNameFormat = acctNameFormat.subString(myMatcherDoubleTokenCheck.start(), myMatcherDoubleTokenCheck.end());
+        } else if (myMatcher.find()) {
+            //If it finds  {!FirstName}
+            firstNameFormat = acctNameFormat.subString(myMatcher.start()-1, myMatcher.end());
+        } else {
+            firstNameFormat = '';
         }
-        //Build prefix
-        namePrefix = setting.substring(0, indexFirstParenthesis);
-        //Build Suffix
-        nameSuffix = setting.substring(indexAfterLastParenthesis+1, setting.length());
-        setting = setting.substring(indexFirstParenthesis, indexAfterLastParenthesis+1);
-
-        //Split strings into a list
-        List<String> settingsSplitSpace= setting.split(' ');
-        for (String settingSplitSpace : settingsSplitSpace) {
-            if (settingSplitSpace.contains('{!') && settingSplitSpace.contains('}')) {
-
-                settingsSplitRightParenthesis = settingSplitSpace.split('}');
-
-                for (String settingSplitRightParenthesis : settingsSplitRightParenthesis) {
-                    if (settingSplitRightParenthesis.contains('{!')) {
-                        settingsSplitLeftParenthesis = settingSplitRightParenthesis.split('\\{!');
-                        for (String settingSplitLeftParenthesis : settingsSplitLeftParenthesis) {
-                            if (UTIL_Describe.isValidField('Contact', settingSplitLeftParenthesis.toLowerCase())) {
-                                setContactFields.add(settingSplitLeftParenthesis.toLowerCase());
-                            }
-                            if (settingSplitLeftParenthesis != '')
-                                listNameStrings.add(settingSplitLeftParenthesis);
+
+        return firstNameFormat;
+    }
+
+    /*******************************************************************************************************
+    * @description Given a NameSpec and a list of Contacts (assumed from a single household),
+    * return the formated Account name.
+    * This can be used for any of the household strings (name, formal greeting, informal greeting).
+    * @param cons is the list of Contacts
+    * @param ns the NameSpec class
+    * return String.
+    */
+    private static String buildAccountName(List<Contact> cons, NameSpec ns) {
+        Set<String> processedLastNames = new Set<String>();
+        List<String> finalAccountName = new String[0];
+        String salutationSpec = '';
+        Integer conSize = cons.size();
+        Boolean isSalutationFormat = ns.salutationSpec == '' ? False : True;
+        Boolean isNoLastnames = ns.fullNameSpec.toUpperCase().indexOf(chLToken + 'LASTNAME' + chRToken) == -1;
+        Boolean isLastNameOnlyFormat = !isNoLastnames && ns.returnNameFormat().size() == 1;
+        Boolean lastNameNotIncludedInFormat = ns.acctNameFormat.indexOf(chLToken + 'LastName' + chRToken) == -1;
```

Comment:
@kdy618 Similar to line 470, should we use the `toUpperCase()` method here in case it's not entered as "LastName" exactly? (Not sure if this matters or not, I just happened to notice it was different...)

Reply:
Also, how is this boolean different than `isNoLastnames`?

Reply:
Finally, is there a unit test that should have been add/modified to verify the latest commit? (https://github.com/SalesforceFoundation/EDA/pull/948/commits/96723b42a98038d55680cb40f06d4c2b0adfd386)

Reply:
@robertbwatson , I've changed the boolean variable names since they are a bit confusing. **isNoLastNames** differ from **lastNameNotIncludedInFormat** because **lastNameNotIncludedInFormat** is checking **acctNameFormat** which contains the actual Account name format selected in the EDA settings while **isNoLastNames** is checking if **fullNameSpec** which may or may not contain the string "_FullNameSpec_" or **acctNameFormat**.  I missed the unit test. I'll be adding it.